### ISSUE TITLE
Error on rubicon tag

### ISSUE THREAD

**Issue by justforfun** [krux/postscribe#93]
I get the following error
Uncaught TypeError: Cannot read property 'SRC' of undefinedhtmlParser.js:362 
when trying to manage with Postscribe this Rubicon script:
http://optimized-by.rubiconproject.com/a/10653/44584/194562-15.js?&cb=0.6909123514778912&tk_st=1&rp_s=c&p_exp=1&p_pos=atf&p_screen_res=1680x1050

The error seems to happen when this doc.write is processed
```javascript
document.write(\'<IFRAME SRC=\"\'+server+\'/partner/scripts/rubicon/emily.html?rtb_ext=1&pc=10653/44584&geo=eu&co=it\" FRAMEBORDER=\"0\" MARGINWIDTH=\"0\" MARGINHEIGHT=\"0\" SCROLLING=\"NO\" WIDTH=\"0\" HEIGHT=\"0\" style=\"height: 0px; width: 0px\"></IFRAME>\');
```
The document.write is part of this function
```javascript
(function() { var s = document.createElement(\"script\"), el = document.getElementsByTagName(\"script\")[0]; s.async = true; s.src = (document.location.protocol == \"https:\" ? \"https://sb\" : \"http://b\") + \".scorecardresearch.com/beacon.js\"; \n   el.parentNode.insertBefore(s, el); \n   })(); \n }\n   \n})();\n<\/script><DIV STYLE=\"height: 0px; width: 0px; overflow: hidden\">\n<script>\n(function() {var proto = document.location.protocol; var server=\"http://tap2-cdn.rubiconproject.com\"; if (proto == \"https:\") server=\"https://tap-secure.rubiconproject.com\"; document.write(\'<IFRAME SRC=\"\'+server+\'/partner/scripts/rubicon/emily.html?rtb_ext=1&pc=10653/44584&geo=eu&co=it\" FRAMEBORDER=\"0\" MARGINWIDTH=\"0\" MARGINHEIGHT=\"0\" SCROLLING=\"NO\" WIDTH=\"0\" HEIGHT=\"0\" style=\"height: 0px; width: 0px\"></IFRAME>\');
```

**Comment by jnewman** [comment]
I agree this seems related. I'm guessing the issue is that `SRC` is caps. There's a workaround for this for scripts (https://github.com/krux/postscribe/blob/master/postscribe.js#L368). We probably need the equivalent for iframes. I can implement later this week

**Comment by justforfun** [comment]
During my tests I also tried to reproduce the problem working on a copy of the Rubicon js.
One test I made was changing the caps of the iframe SRC attribute but the error kept on occuring.

**Comment by jnewman** [comment]
Can you setup a test page on jsfiddle or
http://plnkr.co/ ?

**Comment by justforfun** [comment]
Here it is:
http://jsfiddle.net/w12axrec/

Try to run the script several times  and you'll get the error

**Comment by jnewman** [comment]
Hmm, I'm not seeing it. Which browser are you on? Do you have any plugins?

**Comment by justforfun** [comment]
Sorry I didn't save the last changes I made on jsFiddle.
Here is the test page with the error
http://jsfiddle.net/w12axrec/5/

I ran the test on Chrome, Safari and Firefox getting the following error on each of them:
Firefox
uncaught exception: [object Object]
htmlParser.js:362 TypeError: tok.booleanAttrs is undefined

Chrome
htmlParser.js:362 Uncaught TypeError: Cannot read property 'SRC' of undefined
(index):44 Uncaught #<Object>

Safari
[Error] TypeError: undefined is not an object (evaluating 'tok.booleanAttrs[key]')
    startTag (htmlParser.js, line 362)
    atomicTag (htmlParser.js, line 352)
    tokenToString (htmlParser.js, line 376)
    (funzione anonima) (postscribe.js, line 285)
    each (postscribe.js, line 63)
    buildChunk (postscribe.js, line 283)
    writeStaticTokens (postscribe.js, line 247)
    writeImpl (postscribe.js, line 234)
    write (postscribe.js, line 208)
    write (postscribe.js, line 631)
    write (postscribe.js, line 639)
    (funzione anonima) (_display, line 2)
    global code (_display, line 2)
    replaceChild ([native code], line 0)
    insertScript (postscribe.js, line 543)
    writeScriptToken (postscribe.js, line 506)
    handleScriptToken (postscribe.js, line 388)
    writeImpl (postscribe.js, line 237)
    write (postscribe.js, line 208)
    onScriptDone (postscribe.js, line 476)
    (funzione anonima) (postscribe.js, line 389)
    writeScriptToken (postscribe.js, line 508)
    handleScriptToken (postscribe.js, line 388)
    writeImpl (postscribe.js, line 237)
    write (postscribe.js, line 208)
    write (postscribe.js, line 631)
    write (postscribe.js, line 639)
    global code (194562-15.js, line 2)
[Error] [object Object]
    error (_display, line 44)
    onerror (postscribe.js, line 652)
    replaceChild ([native code], line 0)
    insertScript (postscribe.js, line 543)
    writeScriptToken (postscribe.js, line 506)
    handleScriptToken (postscribe.js, line 388)
    writeImpl (postscribe.js, line 237)
    write (postscribe.js, line 208)
    onScriptDone (postscribe.js, line 476)
    (funzione anonima) (postscribe.js, line 389)
    writeScriptToken (postscribe.js, line 508)
    handleScriptToken (postscribe.js, line 388)
    writeImpl (postscribe.js, line 237)
    write (postscribe.js, line 208)
    write (postscribe.js, line 631)
    write (postscribe.js, line 639)
    global code (194562-15.js, line 2)

**Comment by jnewman** [comment]
Agree this is a bug. I'll take a look tomorrow.

**Comment by justforfun** [comment]
I found that the error occurs when using a previous version of htmlparser.js.
Here the test using the latest version of htmlparser:
http://jsfiddle.net/w12axrec/5/
The error doesn't seem to show up anymore and here the test using the previous version of the library
http://jsfiddle.net/w12axrec/7/
where you can catch the error immediately

**Comment by sethyates** [comment]
We believe this is might be fixed in 2.0.0.  Please help retest against 2.0.0 release and either close or update this ticket with more details.

**Comment by sethyates** [comment]
Cannot reproduce (in 2.0.0).
